assayist/processor/container_rpm_analyzer.py

```diff
@@ -0,0 +1,110 @@
+        build_id = build_info['id']
+        if not self.is_container_build(build_info):
+            log.info(f'Skipping build {build_id} because the build is not a container')
+            return
+
+        self.koji_session = get_koji_session()
+        # Create a mapping of arch to archive (container image) so we can easily map to the
+        # parent container archives in a future loop
+        arch_to_archive = {}
+        not_container_msg = 'Skipping archive {0} since it\'s not a container image'
+        for archive in self.koji_session.listArchives(build_id):
+            if not archive['filename'].startswith('docker-image'):
+                log.debug(not_container_msg.format(archive['id']))
+                continue
+            arch = archive['extra']['image']['arch']
+            if arch in arch_to_archive:
+                log.error(
+                    f'Build {build_id} has more than one container image with the arch {arch}')
+                continue
+            arch_to_archive[arch] = archive
+
+        parent_build_id = build_info['extra']['image'].get('parent_build_id')
+        # If there is a parent to this image, then only get the RPMs installed in this layer
+        # and mark them as embedded artifacts on this container image
+        if parent_build_id is not None:
+            # Find the RPMs installed in this layer versus the parent image
+            for archive in self.koji_session.listArchives(parent_build_id):
+                if not archive['filename'].startswith('docker-image'):
+                    log.debug(not_container_msg.format(archive['id']))
+                    continue
+                arch = archive['extra']['image']['arch']
+                if arch not in arch_to_archive:
+                    log.debug(
+                        f'The parent build {parent_build_id} contains an extra arch of {arch}')
+                    continue
+
+                rpms = self._get_rpms_diff(archive['id'], arch_to_archive[arch]['id'])
+                self._process_embedded_rpms(archive, rpms)
+        # If there is no parent, then this is a base image. Just get all the RPMs installed in
+        # the image and mark them as embedded artifacts in this container image.
+        else:
+            for archive in arch_to_archive.values():
+                rpms = self.koji_session.listRPMs(imageID=archive['id'])
+                self._process_embedded_rpms(archive, rpms)
+
+    def _process_embedded_rpms(self, container_archive, rpms):
+        """
+        Add the nodes and relationships in Neo4j and claim the files these RPMs install.
+
+        :param dict container_archive: the Koji archive of the container these RPMs are embedded in
+        :param list rpms: the list of RPMs that are embedded in the container
+        """
+        artifact_obj = content.Artifact.get_or_create({
+            'archive_id': container_archive['id'],
+            'type_': container_archive['btype']
```

Comment:
The btype will be "image", but don't we use "container" in the DB?

(Minor nit: nice to end final dict/list entries with "," to make future diffs easier to read)

Reply:
Good catch